qiskit/primitives/estimator_result.py

```diff
@@ -41,4 +41,4 @@ class EstimatorResult:
     """
 
     values: "np.ndarray[Any, np.dtype[np.float64]]"
-    metadata: list[dict[str, Any]]
+    metadata: List[Dict[str, Any]]
```

Comment:
`list[dict[str, Any]]` is the better one. The oldest supported Python in qiskit-terra is 3.7 (see https://github.com/Qiskit/qiskit-terra/blob/86fb555871f4811a842380dbf0d54dcda34a4841/setup.py#L81). This means that just using `list` or `dict` works.